### PR TITLE
disable/enable submit button based on upload file validity

### PR DESCRIPTION
**Pull request by jseppi**
Simple disable/enable of the Step 1 form's submit button based on the `changefile` event of the form's file input. The disabling only happens when JS is enabled, so this will degrade well.

The button `disabled` state could use some styling, but that should be taken of by https://github.com/18F/calc/pull/452

### CODE REVIEW

**Comment by toolness** [comment]
Nice! The one thing this might break on, though, are situations where JS is enabled, but the advanced features of the upload widget are *disabled* (aka "degraded") because there's not enough browser support for all its requirements (drag and drop, xhr + FormData, etc).

So you might want to surround your new JS code with a check for whether the upload widget is actually not in degraded mode?

**Comment by toolness** [comment]
I guess an alternative might be to make the upload widget still fire a `changefile` event even when it's in degraded mode... It *might* already do that, but I thought that I had it basically act as though JS didn't exist if not all browser features were present to make it work in a super-awesome way.

My rationale for this was that it's easier to test "no JS at all" and "fully awesome HTML5 browser" than it is to write tests (and UI) for every individual possible HTML5 feature that a browser may or may not have, but that might have been a bad assumption to make...

**Comment by jseppi** [comment]
@toolness: I think I now am taking care of all cases. You are right that `changefile` is not triggered when the upload is degraded. However, the input will still fire its `change` event, so we can listen for that. However again, when the upload is not degraded, the drag-n-drop will not trigger the `change`, but it will trigger `changefile`. So --> we listen for both events and re-enable on either.

**Comment by jseppi** [comment]
Also, when the upload is fully functional, selecting a file through the normal file selection window will cause BOTH `change` and `changefile` to fire, so the handler gets called twice, but that's not a problem since the handler is idempotent.

File: hourglass_site/static_source/js/data-capture/step_1.js [hunk 1]
```diff
@@ -18,13 +18,28 @@ function replaceForm(html) {
...
+  // When we have a fully functional drag-n-drop upload, 'changefile' is
+  // fired on drop, while 'change' is fired on normal file browser selection.
+  // So, we need to listen to both events and reenable the submit button
+  // on either.
+  const enableSubmit = () => $submit.prop('disabled', false);
+  $fileInput.on('changefile', enableSubmit);
+  $fileInput.on('change', enableSubmit);
+
   $(form).on('submit', (e) => {
     const $uploadInput = $('input', $upload);
```
Comment (toolness) [inline hourglass_site/static_source/js/data-capture/step_1.js]: So I think `$uploadInput` is the exact same thing as `$fileInput` now... it'd be cool if we could just use `$fileInput` but I guess we can leave that as a refactoring for later when we do #461!

**Comment by toolness** [comment]
Thanks, this looks great!  Feel free to merge it up.

File: hourglass_site/static_source/js/data-capture/step_1.js (hunk 1 above)
Comment (jseppi) [inline hourglass_site/static_source/js/data-capture/step_1.js]: Oh whoops, didn't notice that. I'll fix real quick then merge.

**Comment by jseppi** [comment]
hrm, my commit message got truncated! It should have read "use $fileInput in place of $uploadInput"

**Pull request merged**